alpaca/rest.go

```diff
@@ -15,33 +15,50 @@ import (
 	"github.com/alpacahq/alpaca-trade-api-go/common"
 )
 
+const (
+	rateLimitRetryCount = 3
+	rateLimitRetryDelay = time.Second
+)
+
 var (
 	// DefaultClient is the default Alpaca client using the
 	// environment variable set credentials
 	DefaultClient = NewClient(common.Credentials())
 	base          = "https://api.alpaca.markets"
-	dataUrl       = "https://data.alpaca.markets"
+	dataURL       = "https://data.alpaca.markets"
 	apiVersion    = "v2"
-	do            = func(c *Client, req *http.Request) (*http.Response, error) {
-		if c.credentials.OAuth != "" {
-			req.Header.Set("Authorization", "Bearer "+c.credentials.OAuth)
-		} else {
-			req.Header.Set("APCA-API-KEY-ID", c.credentials.ID)
-			req.Header.Set("APCA-API-SECRET-KEY", c.credentials.Secret)
-		}
+	do            = defaultDo
+)
+
+func defaultDo(c *Client, req *http.Request) (*http.Response, error) {
+	if c.credentials.OAuth != "" {
+		req.Header.Set("Authorization", "Bearer "+c.credentials.OAuth)
+	} else {
+		req.Header.Set("APCA-API-KEY-ID", c.credentials.ID)
+		req.Header.Set("APCA-API-SECRET-KEY", c.credentials.Secret)
+	}
 
-		resp, err := http.DefaultClient.Do(req)
+	client := &http.Client{
+		Timeout: 10 * time.Second,
```

Comment:
Perhaps the timeout and the retry settings could be configurable w/ sensible defaults